core/modules/layout/layout.module

```diff
@@ -2035,7 +2035,7 @@ function layout_layout_info() {
       'flexible' => TRUE,
       'type' => 'layout',
       'path' => 'templates',
-      'preview' => 'flexible_template.png',
+      'preview' => layout_template_generate_thumbnail($flexible_template),
```

Comment:
Rather than calling this function that generates an image in layout_layout_info(), it would be preferable to do this in the submit handler. It also gives you an opportunity to delete the old image, or generate a new file name.